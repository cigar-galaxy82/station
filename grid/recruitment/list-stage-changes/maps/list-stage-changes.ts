/* eslint-disable jest/no-export, jest/valid-describe, jest/valid-title, jest/no-identical-title */

import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';

import { buildSuperfaceTest } from '../../../test-config';

const describeIf = (
  condition: boolean,
  ...args: Parameters<jest.Describe>
): void => (condition ? describe(...args) : describe.skip(...args));

export const listStageChangesTest = (
  provider: string,
  jobIds: { valid: string },
  candidateIds: { valid: string; invalid: string },
  options?: RecordingProcessOptions
): void => {
  describe(`recruitment/list-stage-changes/${provider}`, () => {
    describe('ListStageChanges', () => {
      let superface: SuperfaceTest;

      beforeAll(() => {
        superface = buildSuperfaceTest({
          profile: 'recruitment/list-stage-changes',
          provider,
          useCase: 'ListStageChanges',
        });
      });

      describe('when specified candidate does exist', () => {
        it('performs correctly', async () => {
          const input =
            provider === 'breezy-hr'
              ? {
                  candidateId: candidateIds.valid,
                  jobId: jobIds.valid,
                }
              : {
                  candidateId: candidateIds.valid,
                };

          const result = await superface.run({ input }, options);

          expect(() => result.unwrap()).not.toThrow();
          expect(result).toMatchSnapshot();
        });
      });

      describe('when specified candidate does not exist', () => {
        it('returns error', async () => {
          const input =
            provider === 'breezy-hr'
              ? {
                  candidateId: candidateIds.invalid,
                  jobId: jobIds.valid,
                }
              : {
                  candidateId: candidateIds.invalid,
                };

          await expect(
            superface.run({ input }, options)
          ).resolves.toMatchSnapshot();
        });
      });

      describeIf(
        provider === 'breezy-hr',
        'when specified company does not exist',
        () => {
          let companyId: string;

          beforeAll(() => {
            companyId = process.env.BREEZY_HR_COMPANY_ID!;

            process.env.BREEZY_HR_COMPANY_ID = '1b111c1111ef11';
          });

          afterAll(() => {
            process.env.BREEZY_HR_COMPANY_ID = companyId;
          });

          it('returns error', async () => {
            const result = await superface.run(
              {
                input: {
                  candidateId: candidateIds.valid,
                  jobId: jobIds.valid,
                },
              },
              options
            );

            expect(() => result.unwrap()).toThrow();
            expect(result).toMatchSnapshot();
          });
        }
      );

      describeIf(
        provider === 'workable',
        'when specified subdomain does not exist',
        () => {
          let subdomain: string;

          beforeAll(() => {
            subdomain = process.env.WORKABLE_SUBDOMAIN!;

            process.env.WORKABLE_SUBDOMAIN = 'invalid-superface';
          });

          afterAll(() => {
            process.env.WORKABLE_SUBDOMAIN = subdomain;
          });

          it('returns error', async () => {
            const result = await superface.run(
              {
                input: {
                  candidateId: candidateIds.valid,
                  jobId: jobIds.valid,
                },
              },
              options
            );

            expect(() => result.unwrap()).toThrow();
            expect(result).toMatchSnapshot();
          });
        }
      );
    });
  });
};
