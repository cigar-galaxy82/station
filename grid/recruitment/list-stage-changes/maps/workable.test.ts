import { listStageChangesTest } from './list-stage-changes';

listStageChangesTest(
  'workable',
  { valid: '2ba8f2' },
  { valid: 'd270c55', invalid: 'NOT-EXISTING' }
);
