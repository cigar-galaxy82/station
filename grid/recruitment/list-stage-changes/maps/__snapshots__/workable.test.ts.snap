// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recruitment/list-stage-changes/workable ListStageChanges when specified candidate does exist performs correctly 1`] = `
Ok {
  "value": Object {
    "changes": Array [
      Object {
        "createdAt": "2022-08-26T03:36:48.612Z",
        "current": false,
        "id": "26e95b86",
        "name": "Applied",
        "stageId": "applied",
      },
      Object {
        "createdAt": "2022-08-30T03:37:28.560Z",
        "current": false,
        "id": "26e95b9a",
        "name": "Phone Screen",
        "stageId": "phone-screen",
      },
      Object {
        "createdAt": "2022-09-03T03:39:36.565Z",
        "current": true,
        "id": "26e95bdb",
        "name": "Assessment",
        "stageId": "assessment",
      },
    ],
    "rateLimit": Object {
      "bucket": "workable",
      "remainingRequests": 9,
      "remainingRequestsPercentage": 90,
      "resetTimestamp": 1671629750,
      "totalRequests": 10,
    },
  },
}
`;

exports[`recruitment/list-stage-changes/workable ListStageChanges when specified candidate does not exist returns error 1`] = `
Err {
  "error": "MappedError: Expected error
Properties: {
  \\"title\\": \\"Not Found\\",
  \\"detail\\": \\"Unable to get candidate by ID NOT-EXISTING. Check that the candidate with that ID exists.\\",
  \\"code\\": \\"NotFound\\",
  \\"rateLimit\\": {
    \\"bucket\\": \\"workable\\"
  }
}
AST Path: definitions[0].statements[1].statements[0]
Original Map Location: Line 16, column 5",
}
`;

exports[`recruitment/list-stage-changes/workable ListStageChanges when specified subdomain does not exist returns error 1`] = `
Err {
  "error": "MappedError: Expected error
Properties: {
  \\"title\\": \\"Not Found\\",
  \\"detail\\": \\"Unable to get subdomain by name invalid-superface. Check that the subdomain with that name exists.\\",
  \\"code\\": \\"NotFound\\",
  \\"rateLimit\\": {
    \\"bucket\\": \\"workable\\"
  }
}
AST Path: definitions[0].statements[0].statements[0]
Original Map Location: Line 9, column 5",
}
`;
